pixels-daemon/src/main/java/io/pixelsdb/pixels/daemon/index/IndexServiceImpl.java

```diff
@@ -120,11 +120,11 @@ public void putIndexEntry(IndexProto.PutIndexEntryRequest request,
         IndexProto.IndexEntry entry = request.getIndexEntry();
 
         // Call SecondaryIndex's putEntry method
-        boolean success = secondaryIndex.putEntry(new SecondaryIndex.Entry(entry.getIndexKey(), 0, entry.getUnique()));
+        long rowId = secondaryIndex.putEntry(new SecondaryIndex.Entry(entry.getIndexKey(), 0, entry.getUnique(), entry.getRowLocation()));
```

Comment:
Try to capture the Exception and set the error code in the response when the exception is caught.

Reply:
Same as in the following code.